File: src/test/java/org/effective_mobile/task_management_system/AssertionsUtils.java
package org.effective_mobile.task_management_system;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;
import org.junit.jupiter.api.Assertions;

import java.util.function.Function;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class AssertionsUtils {
    public static <O, V> void assertEquals(O expected, O actual, Function<O, V> getter) {
        Assertions.assertEquals(getter.apply(expected), getter.apply(actual));
    }

    public static void assertCaselessEquals(String expected, String actual) {
        Assertions.assertEquals(expected.toLowerCase(), actual.toLowerCase());
    }

    public static <E extends Enum<E>> void assertEnumEquals(String expected, E actual) {
        Assertions.assertEquals(expected.toLowerCase(), actual.name().toLowerCase());
    }

    public static <E extends Enum<E>> void assertEnumEquals(E expected, String actual) {
        Assertions.assertEquals(expected.name().toLowerCase(), actual.toLowerCase());
    }
}

File: src/test/kotlin/org/effective_mofile/task_management_system/AssertionsUtils.kt
package org.effective_mofile.task_management_system

import lombok.NoArgsConstructor
import lombok.AccessLevel
import org.junit.jupiter.api.Assertions
import java.util.Locale
import java.util.function.Consumer
import java.util.function.Function

@NoArgsConstructor(access = AccessLevel.PRIVATE)
object AssertionsUtils {
    @JvmStatic
    fun <O, V> assertEquals(expected: O, actual: O, getter: Function<O, V>) {
        Assertions.assertEquals(getter.apply(expected), getter.apply(actual))
    }

    @JvmStatic
    fun <O, V> assertEquals(expected: O, actual: O, getters: List<Function<O, V>>) {
        getters.forEach(Consumer { getter: Function<O, V> -> assertEquals(expected, actual, getter) })
    }

    @JvmStatic
    fun assertCaselessEquals(expected: String?, actual: String?) {
        Assertions.assertEquals(expected?.lowercase(), actual?.lowercase())
    }

    @JvmStatic
    fun <E : Enum<E>> assertEnumEquals(expected: String?, actual: E?) {
        assertCaselessEquals(expected, actual?.name)
    }

    @JvmStatic
    fun <E : Enum<E>> assertEnumEquals(expected: E?, actual: String?) {
        Assertions.assertEquals(expected?.name?.lowercase(), actual?.lowercase())
    }
}